src/crypto/CHIPCryptoPAL.h

```diff
@@ -99,15 +99,45 @@ enum class SupportedECPKeyTypes : uint8_t
     ECP256R1 = 0,
 };
 
+template <typename Sig>
 class ECPKey
 {
 public:
     virtual SupportedECPKeyTypes Type() const = 0;
     virtual size_t Length() const             = 0;
     virtual operator uint8_t *() const        = 0;
+
+    virtual CHIP_ERROR ECDSA_validate_msg_signature(const uint8_t * msg, const size_t msg_length, const Sig & signature) const
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+};
+
+class P256ECDSASignature
+{
+public:
+    size_t & Length() { return length; }
+    size_t Length() const { return length; }
+    operator uint8_t *() const { return (uint8_t *) bytes; }
+
+private:
+    uint8_t bytes[kMax_ECDSA_Signature_Length];
+    size_t length = sizeof(bytes); // kMax_ECDSA_Signature_Length;
```

Comment:
Should this not start at 0 since the data is uninitialized?

Reply:
This function is current also being used to get the capacity of the structure. Maybe it's better to define a separate function to get that. I'll update the code.